### ISSUE TITLE
`<Information> Application: Ready for connections.` is logged before ready for connections

### ISSUE THREAD

**Issue by cjw296** [ClickHouse/ClickHouse#48453]
**Describe what's wrong**

The following message block is logged during server startup:

```
2023.04.05 18:20:36.259522 [ 1 ] {} <Information> Application: Listening for http://0.0.0.0:8123
2023.04.05 18:20:36.259556 [ 1 ] {} <Information> Application: Listening for native protocol (tcp): 0.0.0.0:9000
2023.04.05 18:20:36.259604 [ 1 ] {} <Information> Application: Listening for MySQL compatibility protocol: 0.0.0.0:9004
2023.04.05 18:20:36.259644 [ 1 ] {} <Information> Application: Listening for PostgreSQL compatibility protocol: 0.0.0.0:9005
2023.04.05 18:20:36.259677 [ 1 ] {} <Information> Application: Listening for replica communication (interserver): http://0.0.0.0:9009
2023.04.05 18:20:36.259682 [ 1 ] {} <Information> Application: Ready for connections.
```

However, attempting to connect less than about 3 seconds after that message is logged results in:

```
clickhouse_driver/varint.pyx:62: in clickhouse_driver.varint.read_varint
    ???
clickhouse_driver/bufferedreader.pyx:55: in clickhouse_driver.bufferedreader.BufferedReader.read_one
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   EOFError: Unexpected EOF while reading bytes

clickhouse_driver/bufferedreader.pyx:240: EOFError
```

**Does it reproduce on recent release?**

I am experiencing this in running ` docker.io/clickhouse/clickhouse-server` version version 23.2.1.2537 (official build).

**How to reproduce**

See https://github.com/cjw296/clickhouse-issue-48453.

**Expected behavior**

As soon as `<Information> Application: Ready for connections.` has been logged, I expect to be able to connect an issue statements:

```
    $ python reproducer.py
    2023.04.06 07:22:24.449001 [ 45 ] {} <Information> Application: Ready for connections.
    2023-04-06T07:22:24...: Attempting to execute a statement...
    [('INFORMATION_SCHEMA',), ('default',), ('information_schema',), ('sampledb',), ('system',)]
```

**Error message and/or stacktrace**

```
$ python reproducer.py
2023.04.06 07:22:31.367542 [ 45 ] {} <Information> Application: Ready for connections.
2023-04-06T07:22:31.413540: Attempting to execute a statement...
Error on socket shutdown: [Errno 57] Socket is not connected
Traceback (most recent call last):
  File "/Users/chris/vcs/git/clickhouse-issue-48453/reproducer.py", line 60, in <module>
    main(parser.parse_args().sleep)
  File "/Users/chris/vcs/git/clickhouse-issue-48453/reproducer.py", line 54, in main
    print(client.execute('SHOW DATABASES'))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
File "clickhouse_driver/bufferedreader.pyx", line 240, in clickhouse_driver.bufferedreader.BufferedSocketReader.read_into_buffer
EOFError: Unexpected EOF while reading bytes
```

**Additional context**

I'm building a tool to spin up ephemeral database servers in docker and podman containers for use during development and functional testing. I don't want the tool to have a dependency on database drivers for every type of database it supports, so I rely on the servers accurately logging when they are ready to receive connections. Postgres and MariaDB do this just fine, it would be great if Clickhouse could do the same.

**Comment by filimonov** [comment]
Your reproducer is incorrect. 

Due to the fact that you use 

```
        'CLICKHOUSE_DB': 'sampledb',
```

it start the logic which init some DB objects before giving access to the clickhouse container: 
see  https://github.com/ClickHouse/ClickHouse/blob/86d37916a0cad1cc7d9a441a69a4d39d9907cbb5/docker/server/entrypoint.sh#L107

that logic starts clickhouse w/o full networking (here the 'Ready for connections' is logged the first time), then init stuff, then stop clickhouse, then start it again (with networking, and here 'Ready for connections' is logged the second time). 

Long story short: comment out line `CLICKHOUSE_DB` from your reproducer. OR adjust your `log.find` to find the second occurrence of the 'Ready for connections'.

**Comment by filimonov** [comment]
I'm not sure if we can do something very smart here...

May be we can try to replace clickhouse-server start for init with clickhouse-local with --path=/var/lib/clickhouse ? 

@Felixoid what do you think?

**Comment by cjw296** [comment]
My reproducer is correct in that I _do_ want a test database set up ;-)

However, I hadn't spotted that this line was logged twice in the second case, so my apologies.
I've updated https://github.com/cjw296/clickhouse-issue-48453/commit/10de87976b320c0c38e0608f1aa287f4418f338a to reflect this.

The behaviour, now that I understand it, is in line with postgres, where I need to check for:
```
                b'PostgreSQL init process complete; ready for start up',
                b"LOG:  database system is ready to accept connections",
```

...and mysql, where I need to check for:
```
                b'Temporary server started.',
                b"ready for connections.",
```

So this is fine as it is. My only observation would be that for posgres/mysql, this whole process is <1s, but for Clickhouse it's >3s. However, this is a really minor thing as it's only during testing that these containers are started often.
